java/java.lsp.server/test/unit/src/org/netbeans/modules/java/lsp/server/ui/MockHtmlViewer.java

```diff
@@ -74,6 +73,15 @@ private void load(View view, ClassLoader loader, URL pageUrl, Callable<Object> i
         return clazz.cast(v);
     }
 
+    @Override
+    public <C> C component(View view, Class<C> type) {
+        if (type == Void.class) {
+            load(view);
```

Comment:
Thanks, @dbalek , loading the view after `newView` is finished was one of recent changes. Previously the initialization started sooner then reference to `View` was returned and that lead to various races. Sometimes it worked, sometimes it deadlocked depending on the order of threads execution. Creating the instance first and initializing later when `component(Void.class)` is called fixes these races. I have probably forgotten to update the test - but it actually passed fine this morning (probably too positive order of execution).